contracts/TradeTypesLib2.sol

```diff
@@ -0,0 +1,134 @@
+/*
+ * Hubii Nahmii
+ *
+ * Compliant with the Hubii Nahmii specification v0.12.
+ *
+ * Copyright (C) 2017-2018 Hubii AS
+ */
+
+pragma solidity >=0.4.25 <0.6.0;
+
+import {MonetaryTypesLib} from "./MonetaryTypesLib.sol";
+import {NahmiiTypesLib} from "./NahmiiTypesLib.sol";
+
+/**
+ * @title     TradeTypesLib
+ * @dev       Data types centered around trade
+ */
+library TradeTypesLib2 {
+    //
+    // Enums
+    // -----------------------------------------------------------------------------------------------------------------
+    enum CurrencyRole {Intended, Conjugate}
+    enum LiquidityRole {Maker, Taker}
+    enum Intention {Buy, Sell}
+    enum TradePartyRole {Buyer, Seller}
+
+    //
+    // Structures
+    // -----------------------------------------------------------------------------------------------------------------
+    struct Operator {
+        uint256 id;
+        string data;
+    }
+
+    struct OrderPlacement {
+        Intention intention;
+
+        int256 amount;
+        NahmiiTypesLib.IntendedConjugateCurrency currencies;
+        int256 rate;
+
+        NahmiiTypesLib.CurrentPreviousInt256 residuals;
+    }
+
+    struct OrderParty {
+        uint256 nonce;
+        address wallet;
+
+        NahmiiTypesLib.IntendedConjugateCurrency balances;
+
+        string data;
+    }
+
+    struct Order {
+        OrderParty party;
+
+        OrderPlacement placement;
+
+        NahmiiTypesLib.WalletOperatorSeal seals;
+        uint256 blockNumber;
+
+        Operator operator;
+    }
+
+    struct OrderCancellation {
+        OrderParty party;
+
+        NahmiiTypesLib.WalletOperatorSeal seals;
+        uint256 blockNumber;
+
+        Operator operator;
+    }
+
+    struct TradeOrder {
+        int256 amount;
+        NahmiiTypesLib.WalletOperatorHashes hashes;
+        NahmiiTypesLib.CurrentPreviousInt256 residuals;
+    }
+
+    struct TradeParty {
+        uint256 nonce;
+        address wallet;
+
+        uint256 rollingVolume;
```

Comment:
This is something we don't have for Payments. Do we need this, alternatively, should it have been added for Payments?

Reply:
Rolling volume is specific to trade and to be used to adjust driip fee.

For payments we have argued that rather the size of the amount paid itself should be used to possibly adjust the driip fee. Hence there is no `rollingVolume` counterpart in `PaymentTypesLib.Payment`.